File: Find the Winner of the Circular Game.cpp
class Solution {
public:
    int findTheWinner(int n, int k) {
        vector<int> players;
        for(int i=1; i<=n; i++){
            players.push_back(i);
        }
        int start=0, count=1;
        while(players.size()>1){
            if(count==k){
                players.erase(players.begin()+start);
                count = 0;
                start--;
            }
            start++;
            if(start>=players.size())
                start=0;
            count++;
        }
        return players[0]; 
    }
};

File: Find the Winner of the Circular Game.py
class Solution:
    def findTheWinner(self, n: int, k: int) -> int:
        players = []
        for i in range(1, n+1):
            players.append(i)
        start = 0
        while(len(players) > 1):
            l = len(players)
            players.remove(players[(start+k-1)%len(players)])
            start = (start+k-1)%l

        return players[0]
